document/metadata_test.go

```diff
@@ -0,0 +1,268 @@
+package document
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func validateAnnotation(t *testing.T, as ast.FlatAnnotationsRefSet, want []string) {
+	t.Helper()
+
+	var got []string
+
+	for _, entry := range as {
+		got = append(got, fmt.Sprintf("%s/%s", entry.Annotations.Scope, entry.Annotations.Title))
+	}
+
+	assert.ElementsMatch(t, want, got)
+}
+
+func getTestModules(t *testing.T, modules [][]string) ast.FlatAnnotationsRefSet {
+	t.Helper()
+
+	parsed := make([]*ast.Module, 0, len(modules))
+	for _, entry := range modules {
+		pm, err := ast.ParseModuleWithOpts(entry[0], entry[1], ast.ParserOptions{ProcessAnnotation: true})
+		require.NoError(t, err)
+		parsed = append(parsed, pm)
+	}
+
+	as, err := ast.BuildAnnotationSet(parsed)
+	require.Nil(t, err)
```

Comment:
so far we have followed manual comparisons in testing files, which was also recommended by current maintainers/creators.
Though I'm ok with both, if we end up using it, this needs to be included into `go.mod` as well